src/sentry/consumers/__init__.py

```diff
@@ -162,6 +162,19 @@ def ingest_events_options() -> list[click.Option]:
     return options
 
 
+def ingest_transactions_options() -> list[click.Option]:
+    options = ingest_events_options()
+    options.append(
+        click.Option(
+            ["--no-celery-mode", "no_celery_mode"],
+            default=False,
+            is_flag=True,
+            help="Save event directly in consumer without celery",
+        )
+    )
```

Comment:
My thinking was to roll this out region by region and not percentages. We can rollback the entire region if we think we can't handle the load at any point. I feel like the proliferation of options in the codebase has made things quite difficult to understand.

Reply:
Please reconsider. 

1. we are in a state where a transactions backlog is hard to recover from due to the lack of capacity on Celery. Rolling out with a feature allows you to observe the system on a small percentage of traffic over a period of time with very low risk. 

2. I'd be open to operate this way if you created a new ingest consumer deployment with a small number of replicas (like 1-2) and run only those two in no-celery-mode. Still having an option that everybody can turn off (not only SRE) would be desirable.

3. re: proliferation of options. I find it a feature, not an issue. It means having ways to turn off features without a deployment. Deployment takes hours. We are not able anymore to quickly revert. We need to be able to turn off features instantly when needed.

Reply:
I'm open to adding the option or creating a new consumer deployment when we roll this out to prod, but I don't want to add it now to this PR. For now, this exists only to do load testing in S4S to figure out whether this strategy is even viable, and I would prefer to get started with that now as it will take time. Let's not lock down the SaaS rollout plan now until we have way more data, then decide.

Reply:
Could you please commit to not rolling this out in production outside of sentry for sentry without either adding an option or creating a dedicated deployment with a small number of replicas ?

Reply:
agreed